### PR TITLE
chore(persons): rate limit person destroy endpoint

### PR DESCRIPTION
**Pull request by macobo**
Currently someone is trying to destroy thousands of persons via spamming
the API (see
https://posthog.slack.com/archives/C0185UNBSJZ/p1661887956820539 for
full context). We're reaching out to the client, but adding a rate limit
to this endpoint also makes sense

## How did you test this code?

Manually. Spent quite a bit of time on debugging since current solution doesn't actually rate limit anything.

image

### CODE REVIEW

File: posthog/rate_limit.py [hunk 1]
```diff
@@ -24,3 +24,13 @@ class PassThroughBurstRateThrottle(PassThroughThrottle):
 
 class PassThroughSustainedRateThrottle(PassThroughThrottle):
     scope = "sustained"
+
+
+class DestroyClickhouseModelThrottle(UserRateThrottle):
+    rate = "10/hour"
```
Comment (benjackwhite) [inline posthog/rate_limit.py:30]: Is this delete 10 people per hour? If so that seems overly harsh....

**Review by benjackwhite — COMMENTED** [review]
Noticed this as I had a lot of requests regarding deletions the last weeks. 

Rate limiting sounds good but we shouldn't push users into a stuck place. Are we planning to follow up with a bulk deletion solution?

**Comment by macobo** [comment]
>Are we planning to follow up with a bulk deletion solution?

In abstract? Sure. We need to figure out the UX for that though and someone to build it.

File: posthog/rate_limit.py (hunk 1 above)
Comment (macobo) [reply posthog/rate_limit.py:30]: See PR description - yes, it's harsh, but this is currently causing a lot of pain on cloud.

Comment (pauldambra) [reply posthog/rate_limit.py:30]: Is it worth making the rate into an instance setting so we can change it easily if it's _too_ harsh?

File: posthog/settings/web.py [hunk 2]
```diff
@@ -216,14 +216,19 @@
...
+
+if RATE_LIMIT_ENABLED or TEST:
     # These rate limits are applied to all Django views.
     # Note: Ingestion + decide endpoints do not use Django views, so no rate limits are applied
     REST_FRAMEWORK["DEFAULT_THROTTLE_CLASSES"] = [
         "posthog.rate_limit.PassThroughBurstRateThrottle",
         "posthog.rate_limit.PassThroughSustainedRateThrottle",
     ]
-    REST_FRAMEWORK["DEFAULT_THROTTLE_RATES"] = {"burst": "120/minute", "sustained": "1000/hour"}
+    REST_FRAMEWORK["DEFAULT_THROTTLE_RATES"] = {
```
Comment (pauldambra) [inline posthog/settings/web.py:228]: picking nits... changes to these defaults feels separate. Alters whatever was being tested with the pass through throttles without impacting this change?

**Review by pauldambra — COMMENTED** [review]
couple of fly-by nits :)

File: posthog/settings/web.py (hunk 2 above)
Comment (macobo) [reply posthog/settings/web.py]: Good catch - not intentional at all

File: posthog/rate_limit.py (hunk 1 above)
Comment (macobo) [reply posthog/rate_limit.py:30]: Note that rate limits are only applied on cloud right now.

**Review by yakkomajuri — APPROVED** [review]

File: posthog/rate_limit.py [hunk 3]
```diff
@@ -27,7 +28,7 @@ class PassThroughSustainedRateThrottle(PassThroughThrottle):
 
 
 class DestroyClickhouseModelThrottle(UserRateThrottle):
-    rate = "10/hour"
+    rate = get_instance_setting("DESTROY_CLICKHOUSE_MODEL_THROTTLE_RATE")
```
Comment (macobo) [inline posthog/rate_limit.py]: Doing things this way _always_ causes postgres queries as the app is booting, not when rate limiting is used.

We should avoid this, but I'm looking into fixing the endpoint first.

**Pull request merged**